components/CodeEditor.jsx

```diff
@@ -1,23 +1,37 @@
 import React from "react";
 import CodeContext from "./CodeContext";
-import { useContext, useCallback } from "react";
+import { useContext, useCallback, useEffect } from "react";
 import CodeMirror from "@uiw/react-codemirror";
 import { cpp } from "@codemirror/lang-cpp";
 // import { javascript } from "@codemirror/lang-javascript";
 // import { python } from "@codemirror/lang-python";
 import { aura } from "@uiw/codemirror-theme-aura";
 
-const CodeEditor = () => {
-  const { setCode } = useContext(CodeContext);
-  const onChange = useCallback((value) => {
-    setCode(value);
-  }, []);
+const CodeEditor = ({ problem }) => {
+  const { code, setCode } = useContext(CodeContext);
+
+  useEffect(() => {
+    if (problem) {
+      /* When a problem is selected, set the code editor content to the function header */
+      problem.functionHeader =
+        "def " + problem.title.toLowerCase().replaceAll(" ", "_") + "(nums): ";
+      setCode(problem.functionHeader);
+    }
+  }, [problem, setCode]);
+  const onChange = useCallback(
+    (value) => {
+      setCode(value);
```

Comment:
i didn't think about setting the code context though, check if passing in the boilerplate into value updates this context. if it doesn't then make sure `code` has the boilerplate when they open the problem. just verify it through console logs